example-pallet/src/lib.rs

```diff
@@ -77,7 +77,7 @@ decl_module! {
             T::Currency::transfer(&source, &bridge_id, amount.into(), AllowDeath)?;
 
             let resource_id = T::NativeTokenId::get();
-            <bridge::Module<T>>::transfer_fungible(dest_id, resource_id, recipient, U256::from(amount.saturated_into()))
+            <bridge::Module<T>>::transfer_fungible(dest_id, resource_id, recipient, U256::from(amount.saturated_into::<u128>()))
```

Comment:
I'm not convinced this will always work. There may be cases where BalanceOf represents an unsigned 256-bit integer. Can you try removing the explicit type and confirm that it builds? Seems to work on my end.

Reply:
Hey @ansermino 
yes, its working with nightly-2020-10-06 but throwing error which I mentioned in #85 with '1.52.0-nightly'. 
For now I removed it but I didn't update nightly version.

Reply:
You're right. I must have mixed up my compiler versions. I believe the problem still exists that we want to support 256bit numbers and this is limiting us to 128bits. Any ideas?

Reply:
I think [amount](https://github.com/ChainSafe/chainbridge-substrate/blob/main/example-pallet/src/lib.rs#L18) is already of type u128. Because If you see your runtime :- [Balance](https://github.com/ChainSafe/chainbridge-substrate-chain/blob/master/runtime/src/lib.rs#L57) is of type u128. 
See these links also :-
https://github.com/ChainSafe/chainbridge-substrate-chain/blob/master/runtime/src/lib.rs#L291
https://github.com/ChainSafe/chainbridge-substrate-chain/blob/master/runtime/src/lib.rs#L236
https://github.com/ChainSafe/chainbridge-substrate-chain/blob/master/runtime/src/lib.rs#L57

So basically this `U256::from(amount.saturated_into())`  is converting from u128 to u256.

Reply:
You are right, but I was hoping this could be written in a way to be added to any runtime. By utilizing traits & generics we should be able to make this less restricted. 

Please add the conversion back in and update GH actions to use the more recent compiler. I don't love this solution but this shouldn't block this PR from being merged. Thanks for your patience 🙏

Reply:
Hey @ansermino, can you please create an issue for that or provide more detail for that. I will try to look into it whenever I get time. Meanwhile, for this PR, I am going to do what you said.

Reply:
Thanks for your patience! Will do 😄